framework/DataObjects/DataSet.py

```diff
@@ -567,16 +568,32 @@ def realization(self, index=None, matchDict=None, noMatchDict=None, tol=1e-15, u
           return 0, None
         # otherwise, get it from the collector
         else:
-          index, rlz = self._getRealizationFromCollectorByValue(matchDict, noMatchDict, tol=tol)
+          index, rlz = self._getRealizationFromCollectorByValue(matchDict, noMatchDict, tol=tol, options=options)
       # otherwise, first try to find it in the data
       else:
-        index, rlz = self._getRealizationFromDataByValue(matchDict, noMatchDict, tol=tol, unpackXArray=unpackXArray)
+        index, rlz = self._getRealizationFromDataByValue(matchDict, noMatchDict, tol=tol, unpackXArray=unpackXArray)# should we add options=options to this one as well?
```

Comment:
Address the comment "should we add options=options to this one as well?"

Reply:
This question is to @alfoa